### PR TITLE
Ensure no duplicate ids in instance

### PR DESCRIPTION
**Pull request by fonsecadeline**
To be abble to remove two instances from massive_tests.

### CODE REVIEW

File: models/vrp.rb [hunk 1]
```diff
@@ -136,6 +136,13 @@ def self.create(hash, delete = true)
     end
 
     def self.check_consistency(hash)
+      # ensure IDs consistency
+      # TODO : Active Hash should be checking this
```
Comment (senhalil) [inline models/vrp.rb:140]:
```suggestion
      # ensure IDs are unique
      # TODO: Active Hash should be checking this
```

File: models/vrp.rb [hunk 2]
```diff
@@ -136,6 +136,13 @@ def self.create(hash, delete = true)
...
+      [:matrices, :units, :points, :rests, :zones, :timewindows,
+       :vehicles, :services, :shipments, :subtours].each{ |key|
+        raise OptimizerWrapper::DiscordantProblemError, "#{key} IDs should be uniq" unless hash[key].to_a.collect{ |v| v[:id] }.uniq!.nil?
```
Comment (senhalil) [inline models/vrp.rb:143]:
```suggestion
        raise OptimizerWrapper::DiscordantProblemError, "#{key} IDs should be uniq" unless hash[key].to_a.collect{ |v| v[:id] }.uniq!.nil?
```
```suggestion
        raise OptimizerWrapper::DiscordantProblemError, "#{key} IDs should be unique" unless hash[key].to_a.collect{ |v| v[:id] }.uniq!.nil?
```

**Review by senhalil — APPROVED** [review]
there are some typos otherwise :+1:

File: test/instance_validity_test.rb [hunk 3]
```diff
@@ -93,4 +93,63 @@ def test_assert_inapplicable_for_vroom_if_vehicle_distance
     assert_includes OptimizerWrapper.config[:services][:vroom].inapplicable_solve?(vrp), :assert_no_distance_limitation
     refute_includes OptimizerWrapper.config[:services][:ortools].inapplicable_solve?(vrp), :assert_no_distance_limitation
   end
+
```
Comment (fab-girard) [inline test/instance_validity_test.rb:96]: can you add a functionnal test (in wrapper_test.rb) to check API returns a 400 http status code with duplicated IDs?

**Review by senhalil — APPROVED** [review]

**Review by braktar — APPROVED** [review]

File: models/vrp.rb [hunk 4]
```diff
@@ -136,11 +136,19 @@ def self.create(hash, delete = true)
     end
 
     def self.check_consistency(hash)
+
```
Comment (braktar) [inline models/vrp.rb:139]: Could you remove this extra line ?

**Review by braktar — CHANGES_REQUESTED** [review]

File: test/wrapper_test.rb [hunk 5]
```diff
@@ -3262,4 +3264,13 @@ def test_first_solution_acceptance_with_solvers
     assert_includes OptimizerWrapper.config[:services][:vroom].inapplicable_solve?(vrp), :assert_no_first_solution_strategy
     refute_includes OptimizerWrapper.config[:services][:ortools].inapplicable_solve?(vrp), :assert_no_first_solution_strategy
   end
+
+  def test_reject_when_duplicated_ids
+    vrp = VRP.toy
+    vrp[:services] << vrp[:services].first
+
+    assert_raises OptimizerWrapper::DiscordantProblemError do
+      OptimizerWrapper.wrapper_vrp('demo', { services: { vrp: [:ortools] }}, TestHelper.create(vrp), nil)
```
Comment (braktar) [inline test/wrapper_test.rb:3273]:
```suggestion
      OptimizerWrapper.wrapper_vrp('demo', { services: { vrp: [:demo] }}, TestHelper.create(vrp), nil)
```

**Review by braktar — CHANGES_REQUESTED** [review]

**Review by braktar — APPROVED** [review]

**Review by senhalil — APPROVED** [review]

**Pull request merged**